iocore/aio/AIO.cc

```diff
@@ -223,7 +223,7 @@ aio_init_fildes(int fildes, int fromAPI = 0)
   int i;
   AIO_Reqs *request = (AIO_Reqs *)ats_malloc(sizeof(AIO_Reqs));
 
-  memset(request, 0, sizeof(AIO_Reqs));
+  memset(static_cast<void *>(request), 0, sizeof(AIO_Reqs));
```

Comment:
do we have to do these casts? I thought T* to void* was a non warning event

Reply:
Without the change, I got this:
```
AIO.cc: In function 'AIO_Reqs* aio_init_fildes(int, int)':
AIO.cc:226:38: error: 'void* memset(void*, int, size_t)' clearing an object of type 'struct AIO_Reqs' with no trivial copy-assignment; use value-initialization instead [-Werror=class-memaccess]
   memset(request, 0, sizeof(AIO_Reqs));
```

Reply:
I would guess that ats_malloc() is obsolete.  operator new should throw an exception that will lead to an abort.

If it is not obsolete, why not placement new:
request = new(ats_malloc(sizeof(AIO_Reqs)) AIO_Reqs
?

Reply:
I wouldn't do that on this PR because the point is make it gcc8 friendly. Changing them to placement new should be a separate PR and it should cover all other places.